nemo/collections/llm/api.py

```diff
@@ -95,6 +105,10 @@ def train(
         >>> llm.train(model, data, trainer, tokenizer="data")
         PosixPath('/path/to/log_dir')
     """
+    # [ModelOpt]: If modelopt_state exists, overwrite transformer_layer_spec to modelopt spec
+    if resume and resume.restore_config and resume.restore_config.path:
```

Comment:
If user just passes a path to the checkpoint directly using `AutoResume.resume_from_path` ([here](https://github.com/NVIDIA/NeMo/blob/main/nemo/lightning/resume.py#L72-L73)) this line doesn't get run. the logic should also handle `resume.resume_from_path` 

something like 
```
if resume:
    if resume.restore_config and resume.restore_config.path:
        set_modelopt_spec_if_exists_in_ckpt(model, resume.restore_config.path)
    elif resume.resume_from_path:   
         set_modelopt_spec_if_exists_in_ckpt(model, resume.resume_from_path)
```

other edge case is if user sets `resume_from_directory` in which case there may be multiple checkpoints ... not sure how to best to handle that

Reply:
also you might need to disable gradient accumulation here .. I ran into this error
```
4: [rank4]:   File "/opt/megatron-lm/megatron/core/tensor_parallel/layers.py", line 510, in backward
4: [rank4]:     if weight.main_grad.dtype == torch.float32:
4: [rank4]:        ^^^^^^^^^^^^^^^^
4: [rank4]: AttributeError: 'Tensor' object has no attribute 'main_grad'
```

```
is_quantized_model = set_modelopt_spec_if_exists_in_ckpt(model, resume.restore_config.path)
if is_quantized_model:
            model.config.gradient_accumulation_fusion = False
```